packages/fxa-content-server/server/lib/amplitude.js

```diff
@@ -521,7 +521,7 @@ function receiveEvent(event, request, data) {
     statsd.increment('amplitude.event.raw');
   }
 
-  const userAgent = ua.parse(request.headers['user-agent']);
+  const userAgent = ua.parse(request.headers?.['user-agent']);
```

Comment:
What happens with `userAgent`when `headers` is not defined?  Is it safe to be used later?

It is also extremely unlikely that the there are no headers in the request.  HTTP 1.1 requires a "Host" header.